shim/src/realm.js

```diff
@@ -14,30 +15,20 @@ import { create, defineProperty, defineProperties, freeze, arrayConcat } from '.
 const UnsafeRecForRealmClass = new WeakMap();
 
 function getUnsafeRecForRealmClass(RealmClass) {
-  if (Object(RealmClass) !== RealmClass) {
-    // Detect non-objects.
-    throw new TypeError('internal error: bad object, not a Realm constructor');
```

Comment:
Preventing new Realm() via this side effect felt incorrect. What if there was no weakmap().